pystac_client/client.py

```diff
@@ -443,21 +443,26 @@ def get_collections(self) -> Iterator[Collection]:
                 call_modifier(self.modifier, collection)
                 yield collection
 
-    def get_items(
-        self, *ids: str, recursive: bool | None = None
-    ) -> Iterator["Item_Type"]:
+    def get_items(self, *ids: str, recursive: bool = False) -> Iterator["Item_Type"]:
         """Return all items of this catalog.
 
         Args:
             ids: Zero or more item ids to find.
-            recursive: unused in pystac-client, but needed for falling back to pystac
+            recursive : If True, search this catalog and all children for the
+                item; otherwise, only search the items of this catalog. Defaults
+                to False.
 
         Return:
             Iterator[Item]: Iterator of items whose parent is this
                 catalog.
         """
         if self.conforms_to(ConformanceClasses.ITEM_SEARCH):
-            search = self.search(ids=ids)
+            # Previously, recursive=None was treated the same as recursive=True.
+            # This if statement maintains this behaviour for backwards compatibility.
+            if recursive is not False:
+                search = self.search(ids=ids)
+            else:
+                search = self.search(ids=ids, collections=[self.id])
```

Comment:
This doesn't feel quite right, since the client is a `Catalog`, not a `Collection`.

Reply:
👍🏼 to the docs update. **pystac-client** is for STAC APIs, not static STAC catalogs, and our fallback to **pystac** is more of a convenience than a core feature.

Reply:
Yup, I appreciate the call-out. There's been discussions over the years on whether we should even have the two libraries be separate (for one example, https://github.com/stac-utils/pystac/issues/1334#issuecomment-2082949667). Any documentation cleanup/fixes to make things clearer for folks would be appreciated 🙇🏼.

FWIW My current thinking is that if we ever wanted to go to a v1.0 release of **pystac-client**, we'd want to drop inheritance altogether to avoid these problems.